avl_tree/AvlTree.java

```diff
@@ -0,0 +1,306 @@
+				if (node.right == null)
+					node.hRight = 0;
+				else
+					node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+			}
+			node = isRotate(node);
+		}
+		return node;
+	}
+
+	public NodeAVL rotateLR(NodeAVL node) {
+		NodeAVL sec = node.left;
+		NodeAVL temp = sec.right;
+		node.left = temp;
+		sec.right = temp.left;
+		temp.left = sec;
+		node.left = temp.right;
+		temp.right = node;
+		if (node.left == null)
+			node.hLeft = 0;
+		else
+			node.hLeft = Math.max(node.left.hLeft, node.left.hRight) + 1;
+		if (sec.right == null)
+			sec.hRight = 0;
+		else
+			sec.hRight = Math.max(sec.right.hLeft, sec.right.hRight) + 1;
+		temp.hLeft = Math.max(sec.hLeft, sec.hRight) + 1;
+		temp.hRight = Math.max(node.hLeft, node.hRight) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateRL(NodeAVL node) {
+		NodeAVL sec = node.right;
+		NodeAVL temp = sec.left;
+		node.right = temp;
+		sec.left = temp.right;
+		temp.right = sec;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+		if (sec.left == null)
+			sec.hLeft = 0;
+		else
+			sec.hLeft = Math.max(sec.left.hLeft, sec.left.hRight) + 1;
+		temp.hRight = Math.max(sec.hLeft, sec.hRight) + 1;
+		temp.hLeft = Math.max(node.hLeft, node.hRight) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateLL(NodeAVL node) {
+		NodeAVL temp = node.left;
+		node.left = temp.right;
+		temp.right = node;
+		if (node.left == null)
+			node.hLeft = 0;
+		else
+			node.hLeft = Math.max(node.left.hRight, node.left.hLeft) + 1;
+		temp.hRight = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateRR(NodeAVL node) {
+		NodeAVL temp = node.right;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hRight, node.right.hLeft) + 1;
+		temp.hLeft = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL isRotate(NodeAVL node) {
+		// This Method see if there is nessesity for rotation and if
+		// there is need, it'll do suitable rotation
+		if (node.hRight - node.hLeft >= 2) {
+			if (node.right.hRight - node.right.hLeft >= 1)
+				node = rotateRR(node);
+			else if (node.right.hRight - node.right.hLeft <= -1)
+				node = rotateRL(node);
+		} else if (node.hRight - node.hLeft <= -2) {
+			if (node.left.hRight - node.left.hLeft <= -1)
+				node = rotateLL(node);
+			else if (node.left.hRight - node.left.hLeft >= 1)
+				node = rotateLR(node);
+		}
+		return node;
+	}
+
+	public boolean isEmpty() {
+		return root == null;
+	}
+
+	public void prntIn(NodeAVL node) {
+		if (node == null)
+			return;
+		else if (node.left == null && node.right == null)
+			System.out.print(node.data + " ");
+		else {
+			prntIn(node.left);
+			System.out.print(node.data + " ");
+			prntIn(node.right);
+		}
+	}
+
+	public void del(int data) {

```

Comment:
Just tell me one thing whenever you will use this also, will it be easy to read comments of this method or just knowing it from its name?